subsys/bluetooth/host/iso_internal.h

```diff
@@ -158,3 +158,22 @@ void bt_iso_chan_set_state(struct bt_iso_chan *chan, enum bt_iso_state state);
 
 /* Process incoming data for a connection */
 void bt_iso_recv(struct bt_conn *iso, struct net_buf *buf, uint8_t flags);
+
+#if defined(CONFIG_BT_ISO_UNICAST) || defined(CONFIG_BT_ISO_BROADCASTER)
+
+/* Callback for all ISO SDUs */
+void bt_iso_sent_cb(struct bt_conn *iso, void *user_data, int err);
+
+/* Pull data from the ISO layer */
+struct net_buf *iso_data_pull(struct bt_conn *conn, size_t amount);
+
+/* Returns true if the ISO layer has data to send on this conn */
+bool bt_iso_has_data(struct bt_conn *conn);
+
+#else
+
+#define bt_iso_sent_cb NULL
+#define iso_data_pull(x, y, z) NULL
+#define bt_iso_has_data(x) (void)0
+
+#endif /* defined(CONFIG_BT_ISO_UNICAST) || defined(CONFIG_BT_ISO_BROADCASTER) */
```

Comment:
What's the point of guarding declarations? This prevents us from using `IS_ENABLED`

Reply:
don't remember. I remember fighting with the CI and losing, this is the result.
I'll try again.

Reply:
Seems it's because `iso.c` is not even included in the sources if iso is not enabled.
In that case those fns are undefined, and the linker complains about undefined symbols.

I don't think it should, as I made `is_iso_conn()` et al short-circuiting if `CONFIG_ISO` is not defined. But it does.
Instead, I went full OOP, and am making methods for those 3 functions.

Reply:
Fair. We have several other functions that are not guarded like this and use with `IS_ENABLED`, so not sure what caused the issue you saw. Maybe lack of guards where they were called?